sky/sdk/lib/widgets/focus.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:sky/widgets/widget.dart';
+
+class Focus extends Inherited {
+
+  Focus({
+    GlobalKey key,
+    GlobalKey defaultFocus,
+    Widget child
+  }) : super(key: key, child: child);
+
+  GlobalKey defaultFocus;
+
+  GlobalKey _currentlyFocusedKey;
+  GlobalKey get currentlyFocusedKey {
+    if (_currentlyFocusedKey != null)
+      return _currentlyFocusedKey;
+    return defaultFocus;
+  }
+  void set currentlyFocusedKey(GlobalKey value) {
+    if (value != _currentlyFocusedKey) {
+      _currentlyFocusedKey = value;
+      notifyDescendants();
+    }
+  }
+
+  void syncState(Focus old) {
+    _currentlyFocusedKey = old._currentlyFocusedKey;
+    super.syncState(old);
+  }  
+
+  static bool at(Component component) {
+    assert(component != null);
+    assert(component.key is GlobalKey);
+    Focus focus = component.inheritedOfType(Focus);
+    return focus == null || focus.currentlyFocusedKey == component.key;
```

Comment:
Having it return true means you can add a single focusable input control and you only need to worry about Focus once you add the second one.